packages/inventory-compliance/src/Components/RulesTable/index.js

```diff
@@ -0,0 +1,157 @@
+
+export const selectColumns = (columnTitles) => (
+    columns.filter((column) => columnTitles.includes(column.original || column.title))
+);
+
+export const toRulesArrayWithProfile = (profilesWithRules) => (
+    profilesWithRules.flatMap((profileWithRules) => (
+        profileWithRules.rules.map((rule) => {
+            const identifier = rule.identifier && JSON.parse(rule.identifier);
+            return {
+                ...rule,
+                references: rule.references ? JSON.parse(rule.references) : [],
+                identifier: identifier && identifier.label ? identifier : null,
+                profile: profileWithRules.profile
+            };
+        })
+    ))
+);
+
+const RulesTable = ({
+    system,
+    profileRules,
+    columns,
+    remediationsEnabled = true,
+    selectedFilter,
+    selectable = false
+}) => {
+    const rules = toRulesArrayWithProfile(profileRules);
+    const showPassFailFilter = (columns.filter((c) => (c.title === 'Passed')).length > 0);
+    const { toolbarProps, tableProps, selected: selectedRules } = useTableTools(rules, columns, {
+        identifier: (item) => (
+            `${item.refId}${ item.profile && `-${item.profile.refId}` }`
+        ),
+        selectable: selectable || remediationsEnabled,
+        detailsComponent: RuleDetailsRow,
+        emptyRows: EmptyRows,
+        filterConfig: buildFilterConfig({
+            selectedFilter,
+            showPassFailFilter
+        })
+    });
+
+    let dedicatedAction;
+    if (remediationsEnabled) {
+        const selectedRulesWithRemediations = (selectedRules || []).filter((rule) => (rule.remediationAvailable));
+        dedicatedAction = remediationsEnabled &&
+            <ComplianceRemediationButton
+                allSystems={ [{
+                    id: system.id,
+                    profiles: system.testResultProfiles,
+                    ruleObjectsFailed: [],
+                    supported: system.supported
+                }] }
+                selectedRules={ selectedRulesWithRemediations } />;
+    }
+
+    return <React.Fragment>
+        <PrimaryToolbar { ...toolbarProps } { ...{ dedicatedAction } }/>
+        <Table className='compliance-rules-table' aria-label='Rules table' { ...tableProps }>
+            <TableHeader />
+            <TableBody />
+        </Table>
+        <TableToolbar isFooter className="ins-c-inventory__table--toolbar">
+            <Pagination { ...toolbarProps.pagination } dropDirection='up' variant={ PaginationVariant.bottom } />
+        </TableToolbar>
+    </React.Fragment>;
+};
+
+RulesTable.propTypes = {
+    profileRules: propTypes.array,
+    loading: propTypes.bool,
+    hidePassed: propTypes.bool,
+    system: propTypes.object,
+    remediationsEnabled: propTypes.bool,
+    tailoringEnabled: propTypes.bool,
+    selectedRefIds: propTypes.array,
+    selectedFilter: propTypes.bool,
+    handleSelect: propTypes.func,
+    onSelect: propTypes.func,
+    columns: propTypes.arrayOf(
```

Comment:
Could the columns be refactored to select columns only without any further configuration? It could prevent calling `selectColumns` by the component user.
```
columns: propTypes.arrayOf(propTypes.string)
```